src/dbnode/persist/fs/types.go

```diff
@@ -559,3 +560,11 @@ type NewMergerFn func(
 	contextPool context.Pool,
 	nsOpts namespace.Options,
 ) Merger
+
+// OnDiskSegments represents on index segments on disk for an index volume.
+type OnDiskSegments interface {
```

Comment:
Can we remove all the "OnDisk" and "onDisk" prefixes everywhere? We're already in the "fs" package so "fs.Segments" is a fitting type.